src/Rule/Consignment/DeliveryDateRule.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Sdk\src\Rule\Consignment;
+
+use MyParcelNL\Sdk\src\Model\Consignment\AbstractConsignment;
+use MyParcelNL\Sdk\src\Rule\Rule;
+
+class DeliveryDateRule extends Rule
+{
+    /**
+     * @param  \MyParcelNL\Sdk\src\Model\Consignment\AbstractConsignment $validationSubject
+     */
+    public function validate($validationSubject): void
+    {
+        $isDefaultDeliveryType = AbstractConsignment::DEFAULT_DELIVERY_TYPE !== $validationSubject->getDeliveryType();
+
+        if ($isDefaultDeliveryType && ! $validationSubject->getDeliveryDate()) {
+            $this->addError('If delivery_type is not 2, delivery_date is required.');
```

Comment:
'weten' we hier dat default_delivery_type blijkbaar 2 is? ik zou dan toch de constante sprintf doen voor de duidelijkheid... (als we verder ook nooit '2' gebruiken). verder lijkt het nu alsof $isdefaultdeliverytype nu juist True is als het NIET de default is, lijkt verwarrend.

Reply:
Heb het wat logischer geschreven, variabele naam klopte idd ook niet met wat er nu staat